src/types.ts

```diff
@@ -2,17 +2,23 @@ import { Component, ComponentOptions, Directive, Plugin } from 'vue'
 
 import { DOMWrapper } from './dom-wrapper'
 import { ErrorWrapper } from './error-wrapper'
+import { DOMWrapperError } from './dom-wrapper-error'
 
-export interface WrapperAPI {
+export interface VueWrapperAPI<T> {
+  element: Element
+}
+
+export interface DOMWrapperAPI {
   attributes: (key?: string) => string | Record<string, string>
   classes: (className?: string) => string[] | boolean | ErrorWrapper
   readonly element: Element
   exists: () => boolean
   get<T extends Element>(selector: string): DOMWrapper<T>
-  find<T extends Element>(selector: string): DOMWrapper<T> | ErrorWrapper
+  find<T extends Element>(selector: string): DOMWrapper<T> | DOMWrapperError
```

Comment:
I'm wondering if we should not define all the overload here to be sure that each class properly implement them. Same for get and findAll of course

Reply:
This was what I was thinking too - I will make this change.